cellprofiler/modules/threshold.py

```diff
@@ -783,20 +785,20 @@ def run(self, workspace):
             self.x_name.value, must_be_grayscale=True
         )
         dimensions = input_image.dimensions
-        final_threshold, orig_threshold, guide_threshold = self.get_threshold(
-            input_image, workspace, automatic=False,
+
+        self.final_threshold, self.orig_threshold, self.guide_threshold, binary_image, _ = self.get_threshold(
```

Comment:
Why add these to the class? I'm not against it per se but it's not what we typically do.

Reply:
So the reason for this was tests. Originally, `test_threshold` got the final, local, guide threshold by calling the threshold class method called `get_threshold`. However, when adding threshold to library we wanted more control over which thresholding method you'd like to use, rather than calling this generalised `get_threshold` function. 

So, by adding final_threshold etc. as a class attribute it's possible to get access to these from the tests since now you do not have a `get_threshold` method. Though perhaps there's a better strategy for this.